### DISCUSSION TITLE
Clarity Over Email Addresses

### DISCUSSION THREAD

**Discussion by aljawaid** [creecros/Mailmagik#10] [General]
IF I use webmail, then the email gets picked up

But if I use my mobile, the email wont get sent.  

Same email address (exactly as below):
`Task#635 <help@mymail.com>`

**Comment by aljawaid** [comment]
From mobile, I have tried:
`Task#635<help@mymail.com>`
`Task#635help@mymail.com`

**Reply by aljawaid** [reply]
Just thinking, shouldn't the email address be `+`

`task#365+email@email.com`

**Reply by alfredbuehler** [reply]
Not all servers support this form of addressing.

**Comment by aljawaid** [comment]
![image](https://user-images.githubusercontent.com/10233708/216769355-ffd94025-16d6-4740-8761-0245bfe463d0.png)

Also:
- [x] There is no buttons or links to process that email and convert into task or comment or description etc
- [ ] the dp should be generic if not coming from a user (which most of the time wont be a kanboard user i think)

**Reply by creecros** [reply]
Added a convert to task button, which will convert the email into a task within the same project.
Will do a comment button eventually.

**Reply by creecros** [reply]
I dont know what a dp is. Digital Picture? The Avatar? I kind of like the "?" tbh

**Reply by creecros** [reply]
Well, here is a good question, if we convert an email to a comment....who becomes the commentor? 

I would say, the one converting. Or, maybe the one who sent it, but if they aren't a user in the project, then default to the convertor? Or just allow the non existing user to remain as the comment creator? I mean it doesn't really matter, since you have to be an actual user to perform the action.

For Email to Task, I used the current user who is converting as the creator, that makes sense to me.

**Reply by aljawaid** [reply]
> Well, here is a good question, if we convert an email to a comment....who becomes the commentor?
- I would say the system/Kanboard
 
> I would say, the one converting. Or, maybe the one who sent it, but if they aren't a user in the project, then default to the convertor? Or just allow the non existing user to remain as the comment creator? I mean it doesn't really matter, since you have to be an actual user to perform the action.
- I agree, the converter (with a comment, xx converted this email into a comment/task somewhere for auditing purposes)


> For Email to Task, I used the current user who is converting as the creator, that makes sense to me.

**Reply by aljawaid** [reply]
> I dont know what a dp is. Digital Picture? The Avatar? I kind of like the "?" tbh
- yes, avatar/profile picture... the `?` looks broken... I think it should be 'system/kanboard' so that it shows that it was an automated process, or even an email icon to show the activity/event originated from an email.

**Reply by creecros** [reply]
'system/kanboard' is not a user within kanboard, unfortunately. I'll take a look, but to do something like that, I believe would be quite a bit of extra work, and I'm not seeing the gain. 

xx converted this email, that's simple enough.

Same with the avatar, i'll take a look, but again, i believe we are talking about an extensive amount of work, for little gain. Personally, I like the "?" mark. It might be easy, but just theorizing what that involves, is a lot. Maybe I'm wrong.

**Reply by aljawaid** [reply]
What I mean is that if the user is not recognised then show a generic icon etc

**Reply by aljawaid** [reply]
Xxx converted this email (if user) 

This email was automatically converted into a task {if no user} 

This email was automatically converted into a comment. (if no user)

**Reply by creecros** [reply]
yes, i understand, the ask. generic icon. I'm just saying, its a lot of work. Or better put, now wwe need to override the comment template, redirect to a new avatarhelper, and then we can make a new avatar for generic. Where I'm like, the "?" looks fine to me. That's like rainy day stuff right there.

**Reply by creecros** [reply]
And right now, as it stands, there are no overrides. Which I kind of like. Almost no maintenance probably going forward....

**Reply by creecros** [reply]
1 solution, i just thought of, would be to ask which user in the project to associate the comment to. I actually like this better, no overrides, avatar will always be associated to a user.

**Reply by aljawaid** [reply]
> 1 solution, i just thought of, would be to ask which user in the project to associate the comment to. I actually like this better, no overrides, avatar will always be associated to a user.

Oh yes nice... But default to current logged in user

**Comment by alfredbuehler** [comment]
> IF I use webmail, then the email gets picked up
> 
> But if I use my mobile, the email wont get sent.

Cannot confirm this. Both are working here, FairMail on Android, Roundcube Webmail.

**Comment by creecros** [comment]
So I did it like this:
![image](https://user-images.githubusercontent.com/26339368/217431014-0287e893-a035-4954-bcf1-1ccb46f7d35b.png)

If the email converted to a task was sent by a user in kanboard, then the comment will show as from that user. (not project user restricted, because it can only be converted by a project member anyway)
 - a message included with who converted the email

If the email came from an email address not associated with KB, the one who converted the email will be the comment creator.
- a message included with who sent the original email
